packages/api/src/webhooks/cannon.ts

```diff
@@ -173,6 +174,14 @@ export default class WebhookCannon {
         }),
       };
 
+      // sign payload if there is a webhook secret
+      if (webhook.sharedSecret) {
+        let signature = sign(event.payload, webhook.sharedSecret);
+        params.headers['Livepeer-Signature'] = `t=${Date.now()},v1=${signature}`
```

Comment:
Some considerations:
 - Did you mean to send `params.body` here instead of `event.payload` (this last one is probably `undefined` for most events but `stream.detection`). The `params.body` is the one that is a `string` as well and that would work on the `Buffer.from` call.
 - Should we include the timestamp inside the signed payload? Otherwise, a replay attack could be possible here by re-using the signature but setting new timestamps. I guess the simplest way for that is to include a field in the `params.body` with the timestamp. I think it's a good approach in order to keep a simple interface for signature verification as well.

WDYT?

Reply:
Agree with both points.